build/repo.targets

```diff
@@ -1,56 +1,195 @@
+        _BuildGraphFile=$(_BuildGraphFile);
+        _CloneRepositoryRoot=$(_CloneRepositoryRoot);
+        _DependencyPackagesDirectory=$(_DependencyPackagesDirectory);
+        _RepositoryBuildTargets=$(_RepositoryBuildTargets);
+        _RestoreGraphSpecsDirectory=$(_RestoreGraphSpecsDirectory)" />
+
+    <MSBuild
+      Projects="$(MSBuildThisFileDirectory)PublishPackages.targets"
+      Condition="'$(PublishPackages)'=='true'"
+      Properties="
+        PackagesDirectory=$(ArtifactsDir);
+        DotNetPath=$(DotNetPath);
+        NuGetFeed=$(NuGetPublishFeed);
+        APIKey=$(APIKey)" />
   </Target>
 
-  <Target Name="CIPull">
-    <MSBuild Projects="$(MSBuildProjectFullPath)"
-      Targets="Sake"
-      Properties="SakeTargets=ci-pull;RepositoryRoot=$(RepositoryRoot)" />
+  <Target Name="_GenerateRestoreGraphSpecs" DependsOnTargets="_FindDotNetPath">
+    <ItemGroup>
+      <Solution Include="$(_CloneRepositoryRoot)%(Repository.Identity)\*.sln">
+        <Repository>%(Repository.Identity)</Repository>
+      </Solution>
+
+      <Solution>
+        <AdditionalProperties>RestoreGraphOutputPath=$(_RestoreGraphSpecsDirectory)%(Solution.Repository)\%(Solution.FileName)%(Solution.Extension).json</AdditionalProperties>
+      </Solution>
+    </ItemGroup>
+
+    <MSBuild
+      Projects="@(Solution)"
+      Targets="GenerateRestoreGraphFile"
+      BuildInParallel="$(BuildInParallel)" />
   </Target>
 
-  <Target Name="UpdateRepos">
-    <MSBuild Projects="$(MSBuildProjectFullPath)"
-      Targets="Sake"
-      Properties="SakeTargets=update;RepositoryRoot=$(RepositoryRoot)" />
+  <Target Name="_GenerateBuildGraph" DependsOnTargets="_FindDotNetPath">
+    <Exec
+      Command="$(DotNetPath) run -r &quot;$(_CloneRepositoryRoot) &quot; --graph-specs-root &quot;$(_RestoreGraphSpecsDirectory) &quot; &quot;$(_BuildGraphFile)&quot;"
+      WorkingDirectory="$(RepositoryRoot)tools\BuildGraph\" />
   </Target>
 
-  <Target Name="CleanAll">
-    <MSBuild Projects="$(MSBuildProjectFullPath)"
-      Targets="Sake"
-      Properties="SakeTargets=git-clean;RepositoryRoot=$(RepositoryRoot)" />
+  <Target Name="_UpdateNuGetConfig">
+    <UpdatePackageSource
+      NuGetConfigPath="$(_CloneRepositoryRoot)%(Repository.Identity)\NuGet.config"
+      SourceName="Dependencies"
+      SourceUri="$(_DependencyPackagesDirectory)"
+      Condition="Exists('$(_DependencyPackagesDirectory)')" />
+
+    <UpdatePackageSource
+      NuGetConfigPath="$(_CloneRepositoryRoot)%(Repository.Identity)\NuGet.config"
+      SourceName="Artifacts"
+      SourceUri="$(BuildDir)" />
   </Target>
 
+  <Target Name="_CreateRepositoriesListWithCommits" DependsOnTargets="_FindDotNetPath">
+    <PropertyGroup>
+      <RepositoryFileWithCommit>$(BuildDir)Repositories.props</RepositoryFileWithCommit>
+    </PropertyGroup>
+    <Copy
+      SourceFiles="$(MSBuildThisFileDirectory)Repositories.props"
+      DestinationFiles="$(RepositoryFileWithCommit)" />
+
+    <Exec
```

Comment:
Did you consider using MSBuild instead of a console tool to do this? IIUC what the AddCommitInfo file does, you can probably reduce this to something like:
```xml
<GetGitCommitInfo WorkingDirectory="$(_CloneRepositoryRoot)">
  <Output TaskParameter="CommitHash" PropertyName="_Hash" />
</GetGitCommitInfo>
<XmlPoke2 XmlInputPath="$(RepositoryFileWithCommit)" 
    Query="//Repository[@Include='$(RepoName)']/@Commit" Value="$(_Hash)" />